scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/SpecificationRepository.java

```diff
@@ -3,10 +3,12 @@
 import gov.nasa.jpl.aerie.scheduler.server.exceptions.NoSuchSpecificationException;
 import gov.nasa.jpl.aerie.scheduler.server.models.Specification;
 import gov.nasa.jpl.aerie.scheduler.server.models.SpecificationId;
+import gov.nasa.jpl.aerie.scheduler.server.remotes.postgres.PostgresSpecificationRepository;
 import gov.nasa.jpl.aerie.scheduler.server.services.RevisionData;
 
 public interface SpecificationRepository {
   // Queries
-  Specification getSpecification(SpecificationId specificationId) throws NoSuchSpecificationException;
+  Specification getSpecification(SpecificationId specificationId)
+  throws NoSuchSpecificationException, PostgresSpecificationRepository.GoalBuildFailureException;
```

Comment:
It seems like we probably shouldn't be putting implementation details onto the interface. Could we at least maybe make a more generic exception that the `GoalBuildFailureException` extends? Something like `SpecificationLoadException`?

Reply:
Done.

Reply:
Right - it should be gone now :+1: I got rid of the GoalBuilder-specific exception, and just included a message in the `SpecificationLoadException`. Would it make sense to have both?